packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx

```diff
@@ -52,6 +54,17 @@ export const ConfigurationPage = (props: ConfigurationPageProps): JSX.Element =>
   const errorBarProps = usePropsFor(ErrorBar);
   const adapter = useAdapter();
   const deviceState = adapter.getState().devices;
+
+  let disableStartCallButton = !microphonePermissionGranted || deviceState.microphones?.length === 0;
+  /* @conditional-compile-remove(rooms) */
```

Comment:
Should we create trampoline function to minimize use of conditional-compile statements

Reply:
I think even if I use a trampoline function. I would still have 2 conditional-compile statements. 1 for the `usePermissions` hook and 1 for the the if statement . I cannot have the hook in the function :(.